src/main/scala/streamer/UnrollingAddrGenUnit.scala

```diff
@@ -0,0 +1,112 @@
+package streamer
+
+import chisel3._
+import chisel3.util._
+
+// input and output decleration for unrolling adress generation unit
+// it's a submodule in side data mover
+class UnrollingAddrGenUintIO(
+    unrollingDim: Int = UnrollingAddrGenUintTestParameters.unrollingDim,
+    unrollingFactor: Seq[Int] =
+      UnrollingAddrGenUintTestParameters.unrollingFactor,
+    addrWidth: Int = UnrollingAddrGenUintTestParameters.addrWidth
+) extends Bundle {
+
+  // configurations for unrolling address geenration
+  val valid_i = Input(Bool())
+  val start_ptr_i = Input(UInt(addrWidth.W))
+  val unrollingStrides_i = Input(Vec(unrollingDim, UInt(addrWidth.W)))
+
+  // generated output address
+  val unrolling_addr_o = Output(
+    Vec(unrollingFactor.reduce(_ * _), UInt(addrWidth.W))
```

Comment:
if unrollingFactor = `Seq(8,8)`, does this mean that there are two streamers of 8 ports?
Why do you multiply the elements to generate the number of output addresses. In this case there are 64 32-bit adresses? That seems a lot!

Reply:
> if unrollingFactor = Seq(8,8), does this mean that there are two streamers of 8 ports?

No, it means two nested parfor, each parfor loop bound is 8.

> Why do you multiply the elements to generate the number of output addresses. In this case there are 64 32-bit adresses? That seems a lot!

If the generated signals are not used, aka, not connected to the output, Chisel will optimize it. Here, we generate all unrolled address for expression convenience , but we only use some of them.